mangaki/mangaki/algo/side.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from scipy.sparse import load_npz, issparse
+from sklearn.preprocessing import scale
+import os.path
+
+
+class SideInformation:
```

Comment:
Meh. :/
I'm not alright-alright with this class.
I'd prefer to call it `TagInformation` rather than `SideInformation`.
Also, it depends on the `lasso` folder when it should depend on a `tags` folder.

Reply:
I prefer SideInformation because it matches the research term and it might be used for the AniDB info later, for example (artist, tags, etc.).

Reply:
It's not a matter of being a research term or if it'll be used with AniDB later, the thing now is this class is crafted towards tags processing only, which means:

- If we want to add new "side information", we'll have to rethink all the design of this class.
- This class is just a glorified wrapper around a matrix loader/preprocesser.

I'm not convinced in introducing this as `SideInformation`.

So here is my proposal:

- Either we can generalize this class and call it `SideInformation` but its design will have to change.
- Either we drop the whole class thing and keep it as a `tags_utils.py` or whatever temporarily.

Reply:
I'm okay redesigning it later. So far it is already way better than `illustration2vec.json`. And proofreading @mrsalixor's `anidb_tags_to_json` gave me an idea about how to rewrite it better. So let's merge? 😄

Reply:
:-]
Open an issue then and consider it approved :p.